### PR TITLE
feature: show prices in localized currency

### PR DESCRIPTION
**Pull request by albemarle**
Legacy page:

image

MFE version:

image

### CODE REVIEW

File: src/payment/data/selectors.js [hunk 1]
```diff
@@ -5,18 +5,50 @@ import { configurationSelector } from '../../common/selectors';
...
   configurationSelector,
-  (basket, configuration) => ({
+  localizedCurrencySelector,
+  (basket, configuration, currency) => ({
     ...basket,
     dashboardURL: configuration.LMS_BASE_URL,
     supportURL: configuration.SUPPORT_URL,
     ecommerceURL: configuration.ECOMMERCE_BASE_URL,
     isEmpty: basket.loaded && (!basket.products || basket.products.length === 0),
+    isCurrencyConverted: currency.currencyCode !== 'USD',
```
Comment (albemarle) [inline src/payment/data/selectors.js]: Oops, this is wrong, fix coming up.

Comment (albemarle) [inline src/payment/data/selectors.js]: Oops, this is wrong when the cookie is not set.  Fix coming up.

File: src/payment/CurrencyDisclaimer.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { FormattedMessage, FormattedNumber, injectIntl } from '@edx/frontend-i18n';
+
+import { currencyDisclaimerSelector } from './data/selectors';
+
+function CurrencyDisclaimer(props) {
+  // using FormattedNumber and then appending USD is redundant-- you'll get "$90 USD"-
+  // but that's what the legacy page shows
```
[the remark this answers is not in the document]
Comment (albemarle) [reply src/payment/CurrencyDisclaimer.jsx]: Sure, that's a good point, I'll delete the comment.

File: src/payment/Price.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  const price = (<FormattedNumber
+    value={props.amount}
+    style="currency" // eslint-disable-line react/style-prop-object
+    currency={props.currencyCode}
+  />);
+
+  if (props.showAsLocalizedCurrency) {
+    return (
+      <span >
```
[the remark this answers is not in the document]
Comment (albemarle) [reply src/payment/Price.jsx]: Weird, both ways pass eslint.  Fixed.

File: src/payment/data/reducers.js [hunk 4]
```diff
@@ -50,9 +52,26 @@ const basket = (state = basketInitialState, action = null) => {
...
+      currencyCode: cookie.code,
+      currencyRatePerUSD: cookie.rate,
+    };
+  }
+  return {};
+}
+
+const currencyInitialState = getCurrencyInitialState();
+
+const currency = (state = currencyInitialState) => ({ ...state });
```
[the remark this answers is not in the document]
Comment (albemarle) [reply src/payment/data/reducers.js]: You have to look at it from the inside out.  So the happy owl is sitting in a tree and there are three holes in the tree...  ;-)

File: src/payment/data/selectors.js [hunk 5]
```diff
@@ -5,18 +5,50 @@ import { configurationSelector } from '../../common/selectors';
...
+
+export const basketSummarySelector = createSelector(
+  basketSelector,
+  localizedCurrencySelector,
+  (basket, currency) => {
+    const conversionRate = currency.currencyRatePerUSD || 1;
+
+    return {
+      ...basket,
+      orderTotal: basket.orderTotal * conversionRate,
```
[the remark this answers is not in the document]
Comment (albemarle) [reply src/payment/data/selectors.js]: The legacy page yanks the values out of the elements and shoves the recalculated values back in.  I don't think we should do that.

[the remark this answers is not in the document]
Comment (albemarle) [reply src/payment/data/selectors.js]: (Also, 2 is a good idea.  A LocalizedPrice component could live in frontend-i18n for easy reuse.)

[the remark this answers is not in the document]
Comment (robrap) [reply src/payment/data/selectors.js]: FYI: @AlasdairSwan: Not sure when you will be implementing this in Prospectus.

[the remark this answers is not in the document]
Comment (AlasdairSwan) [reply src/payment/data/selectors.js]: Thanks @robrap, we're actually doing a spike on determining user location in prospectus this sprint (FYI @inventhouse), we've been doing this on edx-mktg for a few years.

[the remark this answers is not in the document]
Comment (robrap) [reply src/payment/data/selectors.js]: Thanks @AlasdairSwan.  At this time, the cookie is the contract, so as long as that gets updated appropriately we won't be breaking things.

**Comment by albemarle** [comment]
Updated for coupons:

localized currencies with coupon

File: src/payment/LocalizedPrice.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  const priceElement = (<FormattedNumber
+    value={price}
+    style="currency" // eslint-disable-line react/style-prop-object
+    currency={props.currencyCode}
+  />);
+
+  if (props.showAsLocalizedCurrency) {
+    return (
+      <span>
+        {priceElement} *
```
Comment (robrap) [inline src/payment/LocalizedPrice.jsx]: Thanks for the refactor.  I think @AlasdairSwan find a way to pull this out and reuse it when he gets there, even with the little *. :)

**Pull request merged**